docs_app/content/tutorial/8-subjects.md

```diff
@@ -0,0 +1,304 @@
+    /** Handle a value notification */
+    notify(value: T): void;
+}
+
+interface OP_Subject<T> {
+    /** Register an observer to be notified by this subject */
+    addObserver(observer: OP_Observer<T>): void;
+    /** Unregister an observer so it stops being notified by this subject */
+    removeObserver(observer: OP_Observer<T>): void;
+    /** Notify all registered observers of a value */
+    notifyObservers(value: T): void;
+}
+```
+
+If you look at the API above, and you squint a little, you'll notice by renaming a few things you'll find some familiar-looking methods in these interfaces. For one, you could rename `notify` to `next` in `OP_Observer`, and you'll have a partial RxJS [`Observer`](API) type. Also, looking at `OP_Subject`, we know of a type that has a method that accepts an observer: [`subscribe`](API) on [`Observable`](API). So if we rename `addObserver` to `subscribe`, we're pretty close. But what about `removeObserver`? That's easy, we can get rid of it entirely, and instead return a [`Subscription`](API) object that, when [unsubscribed](GL), will remove the observer from the subject for us.
+
+That brings us to here:
+
+```ts
+interface OP_Observer<T> {
+    /** Handle a value notification */
+    next(value: T): void;
+}
+
+interface OP_Subject<T> {
+    /** 
+     * Register an observer to be notified by this subject, and return
+     * a Subscription that will remove the observer when you call `unsubscribe()` on it.
+     */
+    subscribe(observer: OP_Observer<T>): Subcription;
+    /** Notify all registered observers of a value */
+    notifyObservers(value: T): void;
+}
+```
+
+That only leaves us with that pesky `notifyObservers` method, which takes a value and returns `void`. WHich happens to be exactly what our `next` method did on `OP_Observer` (and RxJS [`Observer`](API)). So let's change `notifyObservers` to `next`. And, after we factor this out, we have this:
+
+```ts
+interface OP_Observer<T> {
+    /** Handle a value notification */
+    next(value: T): void;
+}
+
+interface OP_Subject<T> extends OP_Observer<T> {
+    /** 
+     * Register an observer to be notified by this subject, and return
+     * a Subscription that will remove the observer when you call `unsubscribe()` on it.
+     */
+    subscribe(observer: OP_Observer<T>): Subcription;
+}
+```
+
+By adding a few additional methods we have exactly the same thing as what we have in RxJS:
+
+```ts
+interface Observer<T> {
+    /** Handle a value notification */
+    next(value: T): void;
+    /** Handle an error notification */
+    error(err: any): void;
+    /** Handle completion */
+    complete(): void;
+}
+
+interface Observable<T> {
+    /** 
+     * Register an observer to be notified, and return
+     * a Subscription that will remove the observer when you call `unsubscribe()` on it.
+     */
+    subscribe(observer: Observer<T>): Subcription;
+}
+
+interface Subject<T> extends Observable<T>, Observer<T> {
+}
+```
+
```

Comment:
The above content puts things in perspective and demonstrates how Observable and Subject are based on well defined design patterns. Cool!